storage/reads/group_resultset.go

```diff
@@ -227,33 +227,30 @@ func (g *groupResultSet) groupBySort() (int, error) {
 	var seriesRows []*SeriesRow
 	vals := make([][]byte, len(g.keys))
 	tagsBuf := &tagsBuffer{sz: 4096}
-	allTime := g.req.Hints.HintSchemaAllTime()
 
 	seriesRow := seriesCursor.Next()
 	for seriesRow != nil {
-		if allTime || g.seriesHasPoints(seriesRow) {
```

Comment:
Why did we allow this previously for `allTime` ? Was there a bug if `allTime` is true but we don't have the `handleRead` change above?

Reply:
Actually I assume such a bug would have to be some race where the series is deleted between the index lookup that finds the series row and the actual read looking for the series. Hopefully that is not possible (if we grab the fileset and the series data while holding read locks...) but might be a good thing to test for if we have similar patterns elsewhere.

Reply:
Also this comment is great:

```
// seriesHasPoints reads the first block of TSM data to verify the series has points for
// the time range of the query.
func (g *groupResultSet) seriesHasPoints(row *SeriesRow) bool {
	// TODO(sgc): this is expensive. Storage engine must provide efficient time range queries of series keys.
```

Reply:
The `allTime` "hint" flag doesn't appear to get set anywhere, nor do any of the other [hint flags](https://github.com/influxdata/influxdb/blob/master/storage/reads/datatypes/hintflags.go), other than in the tests. This seems to concur with a comment in the tests: https://github.com/influxdata/influxdb/blob/75b9398e1a78d5a128ca170094639023795e9de8/storage/reads/group_resultset_test.go#L265-L267

That said, I would surmise that such a flag would be used for specifying that the time range request by the query fully covered all the shard start/stop times. This would allow the logic to short-circuit the more expensive `g.seriesHasPoints` call by assuming that if the series exists on any of the shards and the time range covers all shards, that series must actually contain some data. Using this flag in that way would be a useful partial optimization on its own, however I think it would have limited applicability and it would not be very intuitive that a query with a very large time range was faster than one with a small time range.

There's not a lock covering the index lookup that builds the index series cursor & the data lookup, but if the series data was deleted between those events I think all that should happen is that the series data lookup won't return anything and as such the series won't be included in the result tables. This would be the same thing that would happen after this change: When the shard-level series key list built from the index includes a series key that does not have data for the time range requested by the query, the request for the data from TSM for the series will return an empty list and the series won't be included in the result tables.